src/plugins/application_config/server/string_utils.ts

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { Logger } from 'src/core/server';
+
+const ERROR_MESSSAGE_FOR_EMPTY_INPUT = 'Input cannot be empty!';
+const ERROR_FOR_EMPTY_INPUT = new Error(ERROR_MESSSAGE_FOR_EMPTY_INPUT);
+
+function isEmpty(input: string): boolean {
+  if (!input) {
+    return true;
+  }
+
+  return !input.trim();
+}
+
+export function validate(input: string, logger: Logger) {
```

Comment:
Other than empty checker, could a potential evil regex or unsupported entries (like a made up one) cause trouble on the server side. Or more generally, could a bad config entry mess up the index say` ".opensearch_dashboards_config"` ?

Reply:
Good call out! The APIs I call with the opensearch client are matching document ID exactly. Thus there is no concern for the attack. The `trim` is sufficient. But your comment does remind me to use the trimmed strings when passing to the client. I just made a change accordingly.

In your example of `a bad config entry`, it looks like an empty string which will trigger my current validation logic to throw error. 

I do agree that in the future we could enhance the checks, e.g max length.

Reply:
Added a follow up issue https://github.com/opensearch-project/OpenSearch-Dashboards/issues/6000